doc/developers/contributing.rst

```diff
@@ -810,6 +810,93 @@ To test code coverage, you need to install the `coverage
 
 3. Loop.
 
+Monitoring performance
+======================
+
+*This section is heavily inspired from the* `pandas documentation 
+<https://pandas.pydata.org/docs/development/contributing.html#running-the-performance-test-suite>`_.
+
+
+When proposing changes to the existing code base, it's important to make sure
+that they don't introduce performance regressions. Scikit-learn uses
+`asv benchmarks <https://github.com/airspeed-velocity/asv>`_ to monitor the
+performance of a selection of common estimators and functions. The benchmark
+suite can be found in the `scikit-learn/asv_benchmarks` directory.
+
+To use all features of asv, you will need either `conda` or `virtualenv`. For
+more details please check the `asv installation webpage
+<https://asv.readthedocs.io/en/latest/installing.html>`_.
+
+First of all you need to install the development version of asv::
+
+  pip install git+https://github.com/airspeed-velocity/asv
+
+and change your directory to `asv_benchmarks/`::
+
+  cd asv_benchmarks/
+
+The benchmark suite is configured to run against your local clone of
+scikit-learn. Make sure it is up to date::
+
+  git fetch upstream
+
+In the benchmark suite, the benchmarks are organized following the same
+structure as scikit-learn. For example, you can compare the performance of a
+specific estimator between master and the branch you are working on::
+
+  asv continuous -b LogisticRegression upstream/master HEAD
+
+You can also specify a whole module to benchmark::
+
+  asv continuous -b linear_model upstream/master HEAD
+
+You can replace `HEAD` by any branch. By default it will only report the
+benchmarks that have change by at least 10%. You can control this ratio with
+the `-f` flag.
+
+To run the full benchmark suite, simply remove the `-b` flag ::
+
+  asv continuous upstream/master HEAD
+
+However this can take up to two hours. The `-b` flag also accepts a regular
+expression for a more complex subset of benchmarks to run. The command uses
+conda by default for creating the benchmark environments. If you want to use 
+virtualenv instead, use the `-E` flag::
+
+  asv continuous -E virtualenv upstream/master HEAD
```

Comment:
I think that mentioning `-E virtualenv` would be more useful right after the first command that can potentially fail if conda isn't present, i.e. right after `   asv continuous -b LogisticRegression upstream/master HEAD `. It's also not related to the `-b` flag.